### PR TITLE
Awais786/ecom 3808 Implement Bearer Auth for acceptance tests.

### PR DESCRIPTION
**Pull request by awais786**
Implementing Bearer Authentication.
Update the acceptance test cases.
Add unit test cases.

### CODE REVIEW

**Comment by awais786** [comment]
@ahsan-ul-haq please review this PR.

File: credentials/settings/base.py [hunk 1]
```diff
@@ -288,6 +288,7 @@
 REST_FRAMEWORK = {
     'DEFAULT_AUTHENTICATION_CLASSES': (
         'credentials.apps.api.authentication.JwtAuthentication',
```
Comment (ahsan-ul-haq) [inline credentials/settings/base.py]: replace this with `edx_rest_framework_extensions.authentication.JwtAuthentication`

File: requirements/base.txt [hunk 2]
```diff
@@ -10,6 +10,7 @@ djangorestframework-jwt==1.7.2
 django-rest-swagger==0.3.4
 django-storages==1.1.8
 edx-auth-backends==0.2.3
+edx-drf-extensions==0.5.0
```
Comment (ahsan-ul-haq) [inline requirements/base.txt]: add `edx-drf-extensions==0.5.1`

File: credentials/settings/base.py (hunk 1 above)
Comment (awais786) [inline credentials/settings/base.py]: That should be another story.

File: credentials/apps/api/authentication.py [hunk 3]
```diff
@@ -63,3 +65,16 @@ def authenticate_credentials(self, payload):
...
+    """
+    Simple token based authentication.
+
+    This authentication class is useful for authenticating an OAuth2 access token against a remote
+    authentication provider. Clients should authenticate by passing the token key in the "Authorization" HTTP header,
+    prepended with the string `"Bearer "`.
+    """
+    def get_user_info_url(self):
+        """ Returns the URL, hosted by the OAuth2 provider, from which user information can be pulled. """
+        return '{base}/user_info/'.format(base=settings.OAUTH2_PROVIDER_URL)
```
Comment (ahsan-ul-haq) [inline credentials/apps/api/authentication.py]: I don't think that would be needed.

Comment (awais786) [inline credentials/apps/api/authentication.py]: I just follow the ecommerce way. Now removing this.

**Comment by awais786** [comment]
@rlucioni Please review this PR.

**Comment by rlucioni** [comment]
👍

**Comment by ahsan-ul-haq** [comment]
👍

**Comment by awais786** [comment]
This PR is depending on @tasawernawaz [PR 104](https://github.com/edx/credentials/pull/104/) since credentials block is moved from dashboard to programs tab.

**Pull request merged**